apps/trackers/api.py

```diff
@@ -32,81 +32,89 @@ def post(self, request, *args, **kwargs):
+
+        flaw_uuids = serializer.validated_data["flaw_uuids"]
+
+        # select all active PS modules
+        active_modules = PsModule.objects.filter(
+            active_ps_update_streams__isnull=False
+        ).distinct()
+        active_module_names = active_modules.values_list("name", flat=True)
+
+        # prepare the list of PS modules which do
+        # not allow to create embargoed trackers
+        exclude_private = active_modules.filter(
+            private_trackers_allowed=False
+        ).values_list("name", flat=True)
+
+        # select all affects related to the given flaws
+        selected_affects = Affect.objects.filter(flaw__uuid__in=flaw_uuids)
+        # prepare the list of applicable affects which
+        # 1) has the correct affectedness:resolution
+        # 2) are either public or permit embargoed trackers
+        embargoed_acl = ACLMixin.get_embargoed_acl()
+        affects = selected_affects.filter(
+            Q(
+                affectedness=Affect.AffectAffectedness.AFFECTED,
+                resolution__in=[
+                    Affect.AffectResolution.FIX,
+                    Affect.AffectResolution.DELEGATED,
+                ],
             )
-            not_applicable = selected_affects.difference(affects)
-
-            targets = {}
-            for affect in affects:
-                key = (affect.ps_module, affect.ps_component)
-                impact = max(affect.impact, affect.flaw.impact)
-
-                ps_module = active_modules.get(name=affect.ps_module)
-
-                if (
-                    key in targets
-                    and targets[key].get("impact")
-                    and targets[key]["impact"] >= impact
-                ):
-                    # component is already suggested for a higher impact flaw/affect -- no-op
-                    continue
-
-                offers = {}
-                for stream in ps_module.active_ps_update_streams.all():
-                    offers[stream.name] = {
-                        "ps_update_stream": stream.name,
-                        "selected": False,
-                        "acked": False,
-                        "aus": False,
-                        "eus": False,
-                    }
-                offers = ProductDefinitionRules().file_tracker_offers(
-                    affect, impact, ps_module, offers
-                )
-                targets[key] = {
-                    "affect": affect,
-                    "ps_module": affect.ps_module,
-                    "ps_component": affect.ps_component,
-                    "streams": list(offers.values()),
-                    "impact": impact,
-                    "selected": False,  # each module is deselected by default
-                }
+            | Q(
+                affectedness=Affect.AffectAffectedness.NEW,
+                resolution=Affect.AffectResolution.NOVALUE,
+            ),
+            Q(ps_module__in=active_module_names),
```

Comment:
Since this Q object is in and relationship with the rest, I think it is sufficient to just put in `ps_module__in=active_module_names` and omit the Q object.

Reply:
Makes sense! I will simplify